src/regional_settings.cpp

```diff
@@ -799,12 +857,12 @@ void apply_region_overlay( const JsonObject &jo, regional_settings &region )
 
     load_overmap_feature_flag_settings( jo, region.overmap_feature_flag, false, true );
 
-    load_overmap_forest_settings( jo, region.overmap_forest, false, true );
+    optional( jo, false, "overmap_forest_settings", region.overmap_forest, bogus_forest_id );
+    optional( jo, false, "overmap_ravine_settings", region.overmap_ravine, bogus_ravine_id );
+    optional( jo, false, "overmap_ocean_settings", region.overmap_ocean, bogus_ocean_id );
```

Comment:
Here too.

So when anything tries to use these to generate, because it doesn't have any actual settings object it's associated with, it gives a debugmsg.

Now, if `was_loaded`, the second parameter to `optional` is false, the value stored there is overwritten. If it's true, then the value is left unchanged. I can't tell what the intent is here, but I'd think these should either be mandatory, or `was_loaded` shouldn't always be false.

Reply:
They definitely shouldn't be mandatory. This is an overlay, so it applies on top of an existing regional_settings. If the settings already exist for this element then we don't want the overlay to apply new ones, so they have to be optional.

However unlike the load_region_settings it might be that we need to check if these parameters were previously loaded (they always wouldn't have been, before). I guess I'd do that like:
```c++
optional( jo, region.overmap_forest.was_loaded, "overmap_forest_settings", region.overmap_forest, bogus_forest_id );
```
but I'm not completely sure, I'm rather out of my depth here.

Reply:
Sorry, I wanted to make sure I didn't miss one of the loads, so I split the comment across both.

Here are the questions I have:

1. Is it valid to create a `region_settings` without forest settings, etc?
2. If it is, what should happen? Are there some default settings it should fall back to?

3. For a region settings that doesn't specify forest settings, etc, what should happen? Should it leave it unchanged (for this, `was_loaded` is always true)?

The code you have in your comment doesn't work, because
1. `overmap_forest` is the id of forest settings, and so to access the was_loaded, you need to use the `->` operator.
2. It's a load order bug. If the forest settings load before this, `region.overmap_forest->was_loaded` is true, otherwise it's false.

Reply:
Since this is an overlay, forest settings should always have loaded before this, I believe, because they should have loaded for region_settings - which is working correctly.

if the parent region doesn't specify forest settings, it should use the default object. however, I have less objection to makign these mandatory in the region_settings definition, it's just that those were working fine before with them being optional.

Reply:
> Since this is an overlay, forest settings should always have loaded before this, I believe, because they should have loaded for region_settings - which is working correctly.

Nope. The only way to provide guarantees about things being loaded is by deferring JSON, which as far as I can tell is only really done for things of the same type. You can be lucky with how things are ordered in the filesystem and files, but as soon as someone makes a mod that doesn't do it the same way, it breaks.

> if the parent region doesn't specify forest settings, it should use the default values. however, I have less objection to makign these mandatory in the region_settings definition, it's just that those were working fine before with them being optional.

What are the default values? The ones specified in the struct? There needs to be an object with the default values in the factory, and this id needs to point to that object.